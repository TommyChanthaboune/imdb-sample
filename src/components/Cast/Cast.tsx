import React, { FC } from 'react';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import Avatar from '../../assets/Avatar.svg';

interface ICast {
  name: string;
  character: string;
  portraitPath?: string;
}

export const Cast: FC<ICast> = ({ portraitPath, name, character }) => (
  <div className="cast">
    <div className="cast__portrait">
      {portraitPath ? (
        <LazyLoadImage src={portraitPath} alt={name} />
      ) : (
        <div className="cast__placeholder">
          <LazyLoadImage src={Avatar} alt={name} />
        </div>
      )}
    </div>
    <p className="cast__name">{name}</p>
    <p className="cast__character">{character}</p>
  </div>
);
