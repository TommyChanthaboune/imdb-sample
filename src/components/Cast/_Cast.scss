.cast {
  width: 199px;
  margin: 0 1rem 1rem 0;
  &__portrait {
    height: functions.toRem(232);
    width: functions.toRem(200);
    overflow: hidden;
    img {
      border-radius: functions.toRem(3.73);
    }
  }

  &__name {
    font-size: 18px;
    font-weight: 500;
    margin-bottom: functions.toRem(8);
  }

  &__character {
    color: $gray;
    font-weight: 500;
    margin: 0;
  }

  &__placeholder {
    border-radius: functions.toRem(3.73);
    display: flex;
    align-items: center;
    justify-content: center;
    background: $blue;
    height: 100%;
    width: 100%;
  }
}