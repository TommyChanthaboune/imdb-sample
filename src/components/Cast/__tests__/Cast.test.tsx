import React from 'react';
import { screen, render } from '@testing-library/react';
import { Cast } from '../Cast';

describe('SectionTitle', () => {
  it('should render correctly with portrait', () => {
    const component = render(
      <Cast
        portraitPath="https://a.espncdn.com/combiner/i?img=/i/teamlogos/nfl/500/dal.png&w=100&h=100&transparent=true"
        name="Tommy Chanthaboune"
        character="Developer"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with a placeholder', () => {
    const component = render(
      <Cast name="Tommy Chanthaboune" character="Developer" />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render name and character', () => {
    render(<Cast name="Tommy Chanthaboune" character="Developer" />);
    expect(screen.queryByText('Tommy Chanthaboune')).toBeInTheDocument();
    expect(screen.queryByText('Developer')).toBeInTheDocument();
  });
});
