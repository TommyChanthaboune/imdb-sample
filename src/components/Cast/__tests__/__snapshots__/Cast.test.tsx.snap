// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitle should render correctly with a placeholder 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cast"
      >
        <div
          class="cast__portrait"
        >
          <div
            class="cast__placeholder"
          >
            <img
              alt="Tommy Chanthaboune"
              src="Avatar.svg"
            />
          </div>
        </div>
        <p
          class="cast__name"
        >
          Tommy Chanthaboune
        </p>
        <p
          class="cast__character"
        >
          Developer
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cast"
    >
      <div
        class="cast__portrait"
      >
        <div
          class="cast__placeholder"
        >
          <img
            alt="Tommy Chanthaboune"
            src="Avatar.svg"
          />
        </div>
      </div>
      <p
        class="cast__name"
      >
        Tommy Chanthaboune
      </p>
      <p
        class="cast__character"
      >
        Developer
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SectionTitle should render correctly with portrait 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cast"
      >
        <div
          class="cast__portrait"
        >
          <img
            alt="Tommy Chanthaboune"
            src="https://a.espncdn.com/combiner/i?img=/i/teamlogos/nfl/500/dal.png&w=100&h=100&transparent=true"
          />
        </div>
        <p
          class="cast__name"
        >
          Tommy Chanthaboune
        </p>
        <p
          class="cast__character"
        >
          Developer
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cast"
    >
      <div
        class="cast__portrait"
      >
        <img
          alt="Tommy Chanthaboune"
          src="https://a.espncdn.com/combiner/i?img=/i/teamlogos/nfl/500/dal.png&w=100&h=100&transparent=true"
        />
      </div>
      <p
        class="cast__name"
      >
        Tommy Chanthaboune
      </p>
      <p
        class="cast__character"
      >
        Developer
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
